### ISSUE TITLE
State not being transmitted

### ISSUE THREAD

**Issue by lmulhall-phet** [phetsims/gravity-force-lab#261]
For https://github.com/phetsims/QA/issues/485. Seen on macOS + Chrome 80.0.3987.122.

I have no idea how to reproduce this. (Sorry.)

Somehow, the state was not transmitted:

![issue](https://user-images.githubusercontent.com/31285590/75829650-016d1a80-5d6c-11ea-8164-bb0eaf3447da.png)

**Comment by zepumph** [comment]
The error here is an assertion. In the wrapper code we received a message from the sim that made us think that we had an error, but when we tried to parse the message as an error, it didn't look like an error (i.e. have a message and a stack like all errors should).

I went through the build phet-io.js file from rc.3 and by hand "deminified" the stack shown above it is like so:

```

window.assert
createError: Client.js:1494
createErrors: Client.js:1513
dispatch: Client.js:487

```
From this we can narrow this down to the phet-io frame sending a list of errors in which one or more was a truthy value (because it got past the `if` on line 1493) that didn't have a `stack` and `message` field.

This is strange and a bit worrisome because we should be in complete control of those messages. I went through every spot where we send a response message, and each one sent back an error that was either a ducktyped message/stack object, or `null`. 

I added a better message to the assert in the off chance that we ever reproduce this from master, but I don't know if we will. @lmulhall-phet and I have spent a total of about an hour now trying to reproduce this (I have had fuzzing on this link: `http://localhost/phet-io-wrappers/state/?sim=gravity-force-lab&phetioValidateTandems&phetioDebug&fuzz&numberOfMillisecondsBetweenUpdates=200`) but to no avail. 



OH! While writing this I think I may have figured this out. Perhaps the issue is that and empty string is not truthy, and though the message existed, it was `""`, and so it failed the assertion. I will update the assertion to make sure that this doesn't produce this error.

Note though that we don't know what assertion caused this error. We just know it was in response to a command from the phet-io wrapper.

**Comment by zepumph** [comment]
@samreid will you please review this sneaky type casting bug. I'm a bit worried about how @lmulhall-phet potentially hit an assertion that was masked by this bug, and that we can't reproduce it. That said I think its priority should reflect its difficultly in reproduction. 


I will commit this to the branch, but since there isn't much in the way of reproduce-ability, I don't think QA will need to look over it before publication.

**Comment by zepumph** [comment]
Please assign this back to me so I can add it to the branch.

**Comment by zepumph** [comment]
Here is a patch for a test I was going to add to exercise this problem. It involves triggering an assertion in the sim and trying to catch it back on the sim side. Too late though I realized it was a bad test because assertions always have a message. Something like `Assertion failed: . . . ` 

So I'm even less concerned about this problem now, because it means that somehow a messageless (or stackless) error was thrown in a sim at some point. I looked for usages of `new Error()` and `new Error\( [^'`"]` (regex) and didn't find any usages in our project. I'm not sure where else to go here, and feel good closing, but I want @samreid to review first.

**Comment by zepumph** [comment]
Update. I don't think this is a fuzz reproducealble bug, as I didn't hit this during fuzzing of the state wrapper for 2:15.

**Comment by KatieWoe** [comment]
Managed to reproduce on Edge. Happens if you try to open the about dialog.
![reproducededge](https://user-images.githubusercontent.com/41024075/75910299-e0a7d200-5e0a-11ea-95e6-aa192b970be2.png)

**Comment by KatieWoe** [comment]
This will also cause incorrect launches in studio via the same method.
![preventcorrect](https://user-images.githubusercontent.com/41024075/75912832-0df67f00-5e0f-11ea-9766-f1d42b53227f.gif)

**Comment by zepumph** [comment]
I have not yet been able to reproduce this. Can you do it consistently? Can you do it from master? If you can from master that would be a much nicer stack trace.

**Comment by KatieWoe** [comment]
Consistently yes. Master yes.

**Comment by zepumph** [comment]
After discussing with @KatieWoe, we cannot reproduce on master, but I was able to reproduce on the branch. I then checked out RC shas locally and wasn't able to reproduce unbuilt, but was able to reproduce in a locally built version (2.2.0-rc.3).

**Comment by zepumph** [comment]
I can not reproduce this from a built version on master.

**Comment by zepumph** [comment]
Oh man! What a fun bug!! After more investigation, it looks like this isn't our fault.  But that is jumping ahead. The error, as far as I can see is that in the built/minified version of the sim, spots where we throw "CouldNotYetDeserializeError" instances during state set were not looking like we expected them. We have this catch block in the setState call:

```js

  catch( e ) {
          if ( e instanceof CouldNotYetDeserializeError ) {

            // If the object couldn't be created (if a dependency didn't exist yet), then wait for the next pass.
            nextPass.push( phetioID );
          }
          else { throw e; }
        }

```

And so no `CouldNotYetDeserializeError`s should have been thrown higher that that point, but they were! And in fact they were being sent back to the wrapper as message-less, ambiguous `Error` instances. I think this is because `CouldNotYetDeserializeError` was a subtype that didn't actually add any data, thus the minifier probably just thought it equivalent to `Error`. 

To fix this I added a message to the error (it was previously blank), and updated the catch block to test against the message value too. 

I don't really understand why this wasn't happening on master, but I feel like it could so I committed to master, and we should cherry-pick to the branch.
